.github/workflows/cherry-pick.yml

```diff
@@ -39,4 +39,54 @@ jobs:
           git cherry-pick -x ${{ github.event.inputs.shas }}
           git push --set-upstream https://github.com/ampproject/amphtml.git cherry-pick-${{ github.event.inputs.amp-version }}
 
-      # TODO(estherkim): tie to promote workflow
+  get-channels-to-promote:
+    needs: cherry-pick
+
+    runs-on: ubuntu-latest
+
+    outputs:
+      channels: ${{ steps.get-channels.outputs.channels }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: lts/*
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Get channels with ${{ github.event.inputs.amp-version }}
+        id: get-channels
+        run: |
+          CHANNELS=$(npx ts-node ./scripts/get-channels --amp_version=${{ github.event.inputs.amp-version }})
+          echo "::set-output name=channels::"$CHANNELS""
+
+  promote-beta-opt-in:
+    needs: get-channels-to-promote
+    if: contains(needs.get-channels-to-promote.outputs.channels, 'beta-opt-in')
+    uses: ampproject/cdn-configuration/.github/workflows/promote-beta-experimental-opt-in.yml@main
+    with:
+      amp-version: ${{ github.event.inputs.amp-version }}
+
+  promote-beta-traffic:
+    needs: get-channels-to-promote
+    if: contains(needs.get-channels-to-promote.outputs.channels, 'beta-traffic')
+    uses: ampproject/cdn-configuration/.github/workflows/promote-beta-experimental-traffic.yml@main
+    with:
+      amp-version: ${{ github.event.inputs.amp-version }}
+
+  promote-stable:
+    needs: get-channels-to-promote
+    if: contains(needs.get-channels-to-promote.outputs.channels, 'stable')
+    uses: ampproject/cdn-configuration/.github/workflows/promote-stable.yml@main
+    with:
+      amp-version: ${{ github.event.inputs.amp-version }}
+
+  promote-lts:
+    needs: get-channels-to-promote
+    if: contains(needs.get-channels-to-promote.outputs.channels, 'lts')
+    uses: ampproject/cdn-configuration/.github/workflows/promote-lts.yml@main
+    with:
+      amp-version: ${{ github.event.inputs.amp-version }}
```

Comment:
So I started writing something about potential merge conflicts and how one solution might be to just consolidate all the changes into one PR, but then I realized that this is actually something I prefer to see - I'd rather see one PR titled `🌸 Promoting all 2201011234[000→001] channels` or something like that.

That means creating a new .ts promote file that also takes in a list of channel (or just merge `get-channels.ts` into it). wdyt?

---

Comment I started writing before I pivoted:

I can see these causing a merge conflict, e.g., the first task does:
```diff
 {
-  "beta-opt-in": "032201011234000",
+  "beta-opt-in": "032201011234001",
   "beta-traffic": "032201011234000",
   "control": "022201122239000",
-  "experimental-opt-in": "002201122239000",
+  "experimental-opt-in": "002201122239001",
   "experimental-traffic": "002201122239000",
   "experimentA": null,
```

and the second one does:
```diff
 {
   "beta-opt-in": "032201011234000",
-  "beta-traffic": "032201011234000",
+  "beta-traffic": "032201011234001",
   "control": "022201122239000",
   "experimental-opt-in": "002201122239000",
-  "experimental-traffic": "002201122239000",
+  "experimental-traffic": "002201122239001",
   "experimentA": null,
```

I'm not sure whether Git/GitHub can figure out the correct may to merge two PRs with these changes, so let's try that offline? If it works, yay. If not, we'll have to create a separate promote job that does all the RTV changes together

Reply:
Hmm good foresight. I think you're overthinking this a little. It's ok to have a PR or 2 that will have merge conflicts. It's easy to resolve on the UI. I would rather not create a brand new workflow because it adds complexity and risk for something that doesn't happen often, and is easily resolved when it does.

Reply:
If we're going to have this does automatically for the on-duty, asking them to resolve git merge conflicts is not great. Can you test this on your own repo while also letting it create PRs, then merge them and see if this can be done without having to manually resolve anything?